File: 1996. The Number of Weak Characters in the Game/Solution1996.py
class Solution1996:
    def numberOfWeakCharacters(self, properties: List[List[int]]) -> int:
        properties.sort(key=lambda p:(-p[0], p[1]))
        count,attMax = 0, 0
        for _, d in properties:
            if d < attMax:
                count +=1
            else:
                attMax = max(attMax, d)
        return count

File: 1996. The Number of Weak Characters in the Game/Solution1996.java
class Solution1996 {
    public int numberOfWeakCharacters(int[][] properties) {
        // sorting 
        int n = properties.length;
        int count = 0;
        
        Arrays.sort(properties, (a, b) -> (b[0] == a[0]) ? (a[1] - b[1]) : b[0] - a[0]);
        System.out.println(Arrays.deepToString(properties));
        // 0 - decreasing, (1- increasing)
        int max = 0;
        for (int i = 0; i < n; i++) {
            if (properties[i][1] < max) {
                count++;
            }
            max = Math.max(max, properties[i][1]); // max of 1
        }
        return count;
    }
}